cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/util/Misc.java

```diff
@@ -113,7 +113,8 @@ public static void checkNotNull(Object param, String errorMessagePrefix) throws
      */
     public static void checkNotNullOrEmpty(String param, String errorMessagePrefix) throws
             IllegalArgumentException {
-        // Null is never passed as the parameter here, as per FindBugs and Intellij inspections.
+        checkArgument(!isStringNullOrEmpty(param), (errorMessagePrefix != null ? errorMessagePrefix :
+              "Parameter") + " must not be " + param == null ? "null." : "empty.");
         checkArgument(!isStringNullOrEmpty(param), errorMessagePrefix + " must not be empty.");
```

Comment:
You re-added the old line, but left the new code in place, now we might check twice.